crates/agglayer-certificate-orchestrator/src/network_task.rs

```diff
@@ -255,459 +287,90 @@ where
+                        self.pending_state = Some(new_state);
+                        if let Err(error) = self
+                            .pending_store
+                            .set_latest_proven_certificate_per_network(&self.network_id, &height, &certificate_id)
+                        {
+                            error!(
+                                hash = certificate_id.to_string(),
+                                "Failed to set the latest proven certificate per network: {:?}", error
+                            );
                         }
+                        continue;
                     }
-                    CertificationError::NativeExecutionFailed { source } => {
-                        CertificateStatusError::ProofGenerationError {
-                            generation_type: agglayer_types::GenerationType::Native,
-                            source,
-                        }
-                    }
-                    CertificationError::Types { source } => {
-                        CertificateStatusError::TypeConversionError(source)
-                    }
-
-                    CertificationError::Storage(source) => {
-                        let error_msg = format!(
-                            "Storage error happened in the certification process of \
-                             {certificate_id}",
-                        );
-                        warn!(hash = certificate_id.to_string(), error = ?source, error_msg);
-                        let error = anyhow::Error::from(source).context(error_msg);
-
-                        CertificateStatusError::InternalError(format!("{error:#}"))
-                    }
-                    CertificationError::Serialize { source } => {
-                        let error_msg = format!(
-                            "Serialization error happened in the certification process of \
-                             {certificate_id}",
-                        );
-                        warn!(hash = certificate_id.to_string(), error = ?source, error_msg);
-                        let error = anyhow::Error::from(source).context(error_msg);
-
-                        CertificateStatusError::InternalError(format!("{error:#}"))
-                    }
-                    CertificationError::Deserialize { source } => {
-                        let error_msg = format!(
-                            "Deserialization error happened in the certification process of \
-                             {certificate_id}",
+                    Some(NetworkTaskMessage::CertificateSettled { settled_certificate, .. }) => {
+                        self.at_capacity_for_epoch = true;
+                        self.latest_settled = Some(settled_certificate);
+                        *next_expected_height += 1;
+                        debug!(%certificate_id, "Certification process completed");
+                        let Some(new) = self.pending_state.take() else {
```

Comment:
This should be done before altering the different fields above, right?
If we end up having no state to update in DB, we'll still have the `next_expected_height` updated

Reply:
Maybe you can just move the `let Some(new)...` at the top of the block?